### PR TITLE
Adding type definitions for KalumaJS

### PR DESCRIPTION
**Pull request by jt000**
Creating @types for https://kalumajs.org/docs for developing packages released to a RP2 Pico & Pico W

Going through pages in the docs one by one to construct the typings. The following pages have been completed:
- [X] ADC
- [X] Analog I/O
- [X] AT Command
- [X] Board
- [X] Button
- [X] Console
- [ ] CYW43 - *experimental*
- [ ] Device Driver - *experimental*
- [X] Digital I/O
- [X] Errors
- [X] Events
- [ ] File system - *experimental*
- [ ] Flash
- [X] Global
- [X] GPIO
- [ ] Graphics
- [ ] HTTP - *experimental*
- [x] I2C
- [x] Interrupts
- [x] LED
- [ ] Net - *experimental*
- [ ] Path
- [ ] Process
- [ ] PWM
- [ ] RP2 - *experimental*
- [ ] RTC
- [ ] SDCard
- [ ] SPI
- [ ] Storage
- [ ] Stream - *experimental*
- [X] Timers
- [ ] UART
- [ ] URL
- [ ] WDT
- [ ] Wi-Fi - *experimental*

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/kaluma/tsconfig.json
1. types/kaluma/package.json
1. types/kaluma/tslint.json
1. types/kaluma/index.d.ts
1. types/kaluma/kaluma-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 30dd98f3413528a1d7e781fc741dd1ab467f140e
</p>

File: types/kaluma/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        "@types/kaluma": "workspace:."
+    },
+    "owners": [
+        {
+            "name": "Jason Tremper",
+            "githubUsername": "jt000"
+        },
+        {
+            "name": "Minkyu Lee",
+            "githubUsername": "niklauslee"
```
Comment (peterblazejewicz) [inline types/kaluma/package.json:20]: would be nice to get this user ACK to be included as the owner (I understand this is a lead/one of contributors to a project)

File: types/kaluma/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,1072 @@
+// Kaluma.d.ts
+
```
Comment (peterblazejewicz) [inline types/kaluma/index.d.ts:2]:
```suggestion
```
can mislead (filename)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@jt000 thanks!

File: types/kaluma/package.json (hunk 1 above)
Comment (jt000) [reply types/kaluma/package.json]: Thanks! I'll take them off for now & re-add after I get confirmation

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/kaluma/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 183cae37b6785560a3da7411e4ecd44260cb2a44
</p>

File: types/kaluma/tslint.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (jakebailey) [inline types/kaluma/tslint.json:1]: You should remove this file; it is no longer required.

**Review by jakebailey — CHANGES_REQUESTED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by jt000** [comment]
Ready to merge

**Pull request merged**